src/search/ir.h

```diff
@@ -22,33 +22,49 @@
 
 #include <fmt/format.h>
 
+#include <initializer_list>
 #include <limits>
 #include <memory>
 #include <optional>
 #include <string>
+#include <type_traits>
 #include <utility>
 #include <variant>
 #include <vector>
 
 #include "fmt/core.h"
+#include "ir_iterator.h"
 #include "string_util.h"
+#include "type_util.h"
 
 // kqir stands for Kvorcks Query Intermediate Representation
 namespace kqir {
 
 struct Node {
   virtual std::string Dump() const = 0;
+  virtual std::string_view Name() const = 0;
+  virtual std::string Content() const { return {}; }
```

Comment:
Not sure if my understanding is correct:

`Content` should be always overridden by the leaf node and `ChildBegin` and `ChildEnd` for the non-leaf node like SELECT and HASTAG query?

Reply:
No. `Content` is just to show some additional information about the node, e.g. in `NumericCompareExpr` will show the operator (==/</...).

Reply:
oh? I found some of non-leaf nodes(e.g. TagContainExpr/OrExpr) didn't implement the `Content`, so I thought the `Content` is only meaningful to the leaf node.

Reply:
Because there's no additional information in such node. Also, even for leaf node, if there is no additional information, it's content can also be empty.

Reply:
i see, thanks